src/jdk.hotspot.agent/macosx/native/libsaproc/MacosxDebuggerLocal.m

```diff
@@ -178,6 +184,13 @@ static void throw_new_debugger_exception(JNIEnv* env, const char* errMsg) {
     #define HSDB_FLOAT_STATE        x86_FLOAT_STATE64
     #define HSDB_THREAD_STATE_COUNT x86_THREAD_STATE64_COUNT
     #define HSDB_FLOAT_STATE_COUNT  x86_FLOAT_STATE64_COUNT
+#elif defined(__arm64__)
+    #define hsdb_thread_state_t     arm_thread_state64_t
+    #define hsdb_float_state_t      arm_float_state64_t
+    #define HSDB_THREAD_STATE       ARM_THREAD_STATE64
+    #define HSDB_FLOAT_STATE        ARM_FLOAT_STATE64
```

Comment:
this is wrong, there is not arm_float_state64_t on macos-aarch64
you need to use some neon related 64-bit state here and later in sa-code where you save contexes